tests/test_samplers.cc

```diff
@@ -133,7 +134,54 @@ TEST(test_samplers, test_samplers_gp) {
   const auto ensemble_samples =
       ensemble_sampler(model, dataset, walkers, max_iterations, gen, callback);
 
-  assert(oss->str().size() > 1);
+  EXPECT_GT(oss->str().size(), 1);
+}
+
+inline long int get_group(const double &f) { return lround(floor(f / 5.)); }
```

Comment:
Is `lround` being used to cast from `double` to `long int`? Is there a way that doesn't use two rounding math functions?
This also appears in `test_sparse_gp.cc`

Reply:
we could `static_cast<long int>(floor(f / 5.))` ... but casting from doubles to ints always scares me.

Reply:
I'm pretty sure an integer is guaranteed to have no inaccuracy in `double` representation, because it can be represented exactly, therefore you can safely cast. @jbangelo could you confirm?

Reply:
Casting an `int` to a `double` should be safe on most common platforms since the factional part of a `double` is 52-bits, but casting a `double` to an `int` obviously risks losing precision/data.

`long` is a little more complicated because it tends to be 32 bits on 32 bit platforms, and 64 bits on 64 bit platforms. But casting a `double` to a `long` runs the same risks as with `int`.

It should also be noted that `lround()` won't necessarily clamp the return value if the `double` value is outside the range of `long`.